src/controller/python/chip/crypto/p256keypair.h

```diff
@@ -0,0 +1,99 @@
+                       pychip_P256Keypair_ECDSA_sign_msg aSignMsgFunct, pychip_P256Keypair_ECDH_derive_secret aDeriveSecretFunct);
+    ~pychip_P256Keypair() override;
+
+    CHIP_ERROR Initialize(Crypto::ECPKeyTarget key_target) override;
+
+    /**
+     * @brief Serialize the keypair.
+     * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR Serialize(Crypto::P256SerializedKeypair & output) const override;
+
+    /**
+     * @brief Deserialize the keypair.
+     * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR Deserialize(Crypto::P256SerializedKeypair & input) override;
+
+    /**
+     * @brief Generate a new Certificate Signing Request (CSR).
+     * @param csr Newly generated CSR in DER format
+     * @param csr_length The caller provides the length of input buffer (csr). The function returns the actual length of generated
+     *CSR.
+     * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR NewCertificateSigningRequest(uint8_t * csr, size_t & csr_length) const override;
+
+    /**
+     * @brief A function to sign a msg using ECDSA
+     * @param msg Message that needs to be signed
+     * @param msg_length Length of message
+     * @param out_signature Buffer that will hold the output signature. The signature consists of: 2 EC elements (r and s),
+     * in raw <r,s> point form (see SEC1).
+     * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR ECDSA_sign_msg(const uint8_t * msg, size_t msg_length, Crypto::P256ECDSASignature & out_signature) const override;
+
+    /**
+     * @brief A function to derive a shared secret using ECDH
+     *
+     * This implements the CHIP_Crypto_ECDH(PrivateKey myPrivateKey, PublicKey theirPublicKey) cryptographic primitive
+     * from the specification, using this class's private key from `mKeypair` as `myPrivateKey` and the remote
+     * public key from `remote_public_key` as `theirPublicKey`.
+     *
+     * @param remote_public_key Public key of remote peer with which we are trying to establish secure channel. remote_public_key is
+     * ASN.1 DER encoded as padded big-endian field elements as described in SEC 1: Elliptic Curve Cryptography
+     * [https://www.secg.org/sec1-v2.pdf]
+     * @param out_secret Buffer to write out secret into. This is a byte array representing the x coordinate of the shared secret.
+     * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR ECDH_derive_secret(const Crypto::P256PublicKey & remote_public_key,
+                                  Crypto::P256ECDHDerivedSecret & out_secret) const override;
+
+    /** @brief Return public key for the keypair.
+     **/
+    const Crypto::P256PublicKey & Pubkey() const override
+    {
+        // The mPublicKey is a member of Crypto::P256Keypair and is set in Initialize
+        return mPublicKey;
```

Comment:
This will not work unless we call `mGetPublicKeyFunct`. The internal public key in the base class is likely to be fully out of sync. Suggest building a `P256PublicKey` from the output of `mGetPublicKeyFunct`

Reply:
This is quite tricky, since the signature is `const Crypto::P256PublicKey &` const, we cannot actually change the value.

I think we have to add a UpdatePubkey() and the interface should call it when the key in the python is updated.